Testscripts/Windows/KVP-TestRescind.ps1

```diff
@@ -50,59 +50,58 @@ function Main {
         }
     }
 
-    $checkVM = Check-Systemd -Ipv4 $Ipv4 -SSHPort $VMPort -Username $VMUserName `
-                    -Password $VmPassword
+    $checkVM = Check-Systemd -Ipv4 $Ipv4 -SSHPort $VMPort -Username $VMUserName -Password $VmPassword
     if ( -not $checkVM[-1]) {
         Write-LogInfo "Systemd is not being used. Test Skipped"
-        return "FAIL"
+        return "SKIPPED"
     }
 
     # Get KVP Service status
-    $gsi = Get-VMIntegrationService -Name "Key-Value Pair Exchange" -VMName $VMName -ComputerName $hvServer
+    $gsi = Get-VMIntegrationService -Name $global:VMIntegrationKeyValuePairExchange -VMName $VMName -ComputerName $hvServer
     if ($? -ne "True") {
-        Write-LogInfo "Error: Unable to get Key-Value Pair status on $VMName ($hvServer)"
+        Write-LogErr "Unable to get Key-Value Pair status on $VMName ($hvServer)"
         return "FAIL"
     }
 
     # Check if VM is RedHat 7.3 or later (and if not, check external LIS exists)
-    $supportkernel = "3.10.0.514" #kernel version for RHEL 7.3
-    $null = .\Tools\plink.exe -C -pw $VMPassword -P $VMPort $VMUserName@$ipv4 "yum --version 2> /dev/null"
-    if ($? -eq "True") {
+    if(@("REDHAT", "ORACLELINUX", "CENTOS").contains($global:detectedDistro)) {
+        $supportkernel = "3.10.0.514" #kernel version for RHEL 7.3
         $kernelSupport = Get-VMFeatureSupportStatus -Ipv4 $ipv4 -SSHPort $VMPort -UserName $VMUserName `
                             -Password $VMPassword -SupportKernel $supportkernel
         if ($kernelSupport -ne "True") {
-            Write-LogInfo "Info: Kernels older than 3.10.0-514 require LIS-4.x drivers."
-            $null = .\Tools\plink.exe -C -pw $VMPassword -P $VMPort $VMUserName@$ipv4 `
-                                "rpm -qa | grep kmod-microsoft-hyper-v && rpm -qa | grep microsoft-hyper-v"
-            if ($? -ne "True") {
-                Write-LogInfo "Error: No LIS-4.x drivers detected. Skipping test."
+            Write-LogInfo "Kernels older than 3.10.0-514 require LIS-4.x drivers."
+            $cmd = "rpm -qa | grep kmod-microsoft-hyper-v && rpm -qa | grep microsoft-hyper-v"
+            Run-LinuxCmd -username $VMUserName -password $VMPassword -ip $ipv4 -port $VMPort `
```

Comment:
Receive the Run-LinuxCmd output in $null variable, if you do not want to use. The non-captured output may end up returning to the test controller.